### PR TITLE
fix and reinstate the ignored dns/permissioning/allowlist test 

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

Test was failing because enode containing IP was being resolved via DNS to a hostname, but assertion was still checking for IP. 

Added tests for allowlist behaviour with DNS enabled:
* add enode with IP and then same enode with hostname throws exception
* add enode with hostname should "just work" 
* add enode with IP will resolve to DNS hostname. So check with this behaviour by creating the enodeUrl with DNS config

debug logs from CI
```
03:56:50.876 | Test worker | INFO  | AllowlistWithDnsEnodePersistorAcceptanceTest | enode 2 string enode://5f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@192.168.0.10:1234
03:56:50.916 | Test worker | INFO  | AllowlistWithDnsEnodePersistorAcceptanceTest | enodeURL 2 with DNS enabled but NOT dns update enode://5f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@192.168.0.10:1234
03:56:50.934 | Test worker | INFO  | AllowlistWithDnsEnodePersistorAcceptanceTest | enodeURL 2 with DNS enabled AND dns update enode://5f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@ip-192-168-0-10.ec2.internal:1234
```

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/permissioning/AllowlistWithDnsPersistorAcceptanceTest.java [hunk 1]
```diff
@@ -67,22 +71,48 @@ public void setUp() throws Exception {
...
+
+  @Test
+  public void manipulatedNodesAllowlistWithHostnameShouldWorkWhenDnsEnabled() {
+
+    node.verify(perm.addNodesToAllowlist(ENODE_LOCALHOST_DNS, ENODE_TWO_IP));
+    // use DNS config to resolve the Enode with IP. It either resolves to a hostname or remain as is
+    final EnodeURL enodeURL0 =
+        EnodeURLImpl.fromString(
+            ENODE_TWO_IP,
+            ImmutableEnodeDnsConfiguration.builder().dnsEnabled(true).updateEnabled(true).build());
```
Comment (garyschulte) [inline acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/permissioning/AllowlistWithDnsPersistorAcceptanceTest.java:107]: is this for branch coverage of resolving an enode IP address with name based resolver?

**Review by garyschulte — APPROVED** [review]
LGTM, just a clarifying question

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/permissioning/AllowlistWithDnsPersistorAcceptanceTest.java (hunk 1 above)
Comment (macfarla) [reply acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/permissioning/AllowlistWithDnsPersistorAcceptanceTest.java:107]: yep I added some debug logs to the ticket above. 
it was being resolved to 
`enode://5f8a80d14311c39f35f516fa66...92a0@ip-192-168-0-10.ec2.internal:1234`

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/permissioning/AllowlistWithDnsPersistorAcceptanceTest.java [hunk 2]
```diff
@@ -67,22 +71,48 @@ public void setUp() throws Exception {
     cluster.start(this.node);
   }
 
-  @Ignore("test is failing in CI")
   @Test
-  public void manipulatedNodesAllowlistWithHostnameShouldWorkWhenDnsEnabled() {
+  public void addingEnodeWithIp_andThenAddingSameEnodeWithHostname_shouldThrow() {
```
Comment (pinges) [inline acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/permissioning/AllowlistWithDnsPersistorAcceptanceTest.java:75]: What about the case where add with hostname first and then with IP? Do we have a test?

**Pull request merged**